dotCMS/src/main/java/com/dotmarketing/viewtools/LanguageWebAPI.java

```diff
@@ -168,20 +166,33 @@ public String get(String key, List args) {
 	public String get(String key, String languageId) {
 		String value = null;
 		try {
-			Language lang = LanguageUtil.getUserLanguage(langAPI.getLanguage(languageId), request.getLocale());
-			value = langAPI.getStringKey(lang, key);
+
+			//Find the requested language
+			Language requestedLanguage;
+			try {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)